### PR TITLE
refactor(theme): disable @vue-storefront/core $vsf context plugin

### PR DESCRIPTION
**Pull request by ghost**
see https://github.com/vuestorefront/vue-storefront/pull/6641/
the abillity to do this is new in @vue-storefront/core@2.5.6
(version bump made earlier today)

### CODE REVIEW

**Review by bartoszherba — DISMISSED** [review]

File: packages/theme/components/Header/SearchBar/SearchBar.vue [hunk 1]
```diff
@@ -54,13 +54,10 @@ import {
   defineComponent, ref, watch, useRoute,
 } from '@nuxtjs/composition-api';
 import debounce from 'lodash.debounce';
-import {
-  categoryGetters,
-  useCategorySearch,
-  useFacet,
-} from '@vue-storefront/magento';
+import { categoryGetters } from '@vue-storefront/magento';
```
Comment (bartoszherba) [inline packages/theme/components/Header/SearchBar/SearchBar.vue:57]: We already have getters in the theme package

File: packages/composables/nuxt/plugin.js [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import { integrationPlugin } from '@vue-storefront/core'
+import { integrationPlugin } from '@vue-storefront/magento-theme/helpers/integrationPlugin'
```
Comment (bartoszherba) [inline packages/composables/nuxt/plugin.js:1]: Why did you do that?
Does it couple compsables -> theme instead of composables -> core?

File: packages/theme/modules/context/plugins/context.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
+/**
+ * VSF context from `@vuestorefront/core` is disabled in nuxt.config.js as part of the migration away from core
```
Comment (bartoszherba) [inline packages/theme/modules/context/plugins/context.js:2]: Can we convert this comment into a task so we will not forget to do that? Since only you were working in this area..

**Review by bartoszherba — CHANGES_REQUESTED** [review]

File: packages/theme/components/Header/SearchBar/SearchBar.vue (hunk 1 above)
Comment (sethidden) [reply packages/theme/components/Header/SearchBar/SearchBar.vue]: e5b60820

File: packages/theme/modules/context/plugins/context.js (hunk 3 above)
Comment (sethidden) [reply packages/theme/modules/context/plugins/context.js]: https://vsf.atlassian.net/browse/M2-328
5b5b5158

File: packages/composables/nuxt/plugin.js (hunk 2 above)
Comment (sethidden) [reply packages/composables/nuxt/plugin.js:1]: > Does it couple compsables -> theme instead of composables -> core?
Yeah, this file will be probably moved to theme as part of https://vsf.atlassian.net/browse/M2-288

File: packages/theme/helpers/integrationPlugin/context.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
+/**
```
Comment (Frodigo) [inline packages/theme/helpers/integrationPlugin/context.ts:1]: Can you move this helper to `packages/theme/modules/context` ?

File: packages/theme/helpers/integrationPlugin/index.ts [hunk 5]
```diff
@@ -0,0 +1,55 @@
+import { Context as NuxtContext, Plugin as NuxtPlugin } from '@nuxt/types';
```
Comment (Frodigo) [inline packages/theme/helpers/integrationPlugin/index.ts:1]: Can you move this helper to `packages/theme/modules/context` ?

File: packages/theme/package.json [hunk 6]
```diff
@@ -11,7 +11,7 @@
     "build": "nuxt build -m",
     "build:analyze": "nuxt build -a -m",
     "dev": "nuxt",
-    "dev-debug": "nuxt --inspect",
+    "dev:debug": "node --inspect ../../node_modules/.bin/nuxt dev",
```
Comment (Frodigo) [inline packages/theme/package.json:14]: What is the purpose of this change?

**Review by RetroModernDev — CHANGES_REQUESTED** [review]

File: packages/theme/helpers/integrationPlugin/index.ts (hunk 5 above)
Comment (sethidden) [reply packages/theme/helpers/integrationPlugin/index.ts:1]: Hmm, why though? It's the nuxt types, this doesn't come from core

File: packages/theme/package.json (hunk 6 above)
Comment (sethidden) [reply packages/theme/package.json:14]: It's an additional thing to make debugging easier. There's no `--inspect` flag with in in the `nuxt` command, it's just ignored. i.e. launching `nuxt --inspect` doesn't make any remote debugging server show up in chrome, while launching it with node does.

It's still super wonky as breakpoints stop in a huge server.js file thats +30K lines but at least they stop

File: packages/theme/helpers/integrationPlugin/index.ts (hunk 5 above)
Comment (sethidden) [reply packages/theme/helpers/integrationPlugin/index.ts:1]: Ah, sorry, you mean the whole helper, not the types

File: packages/theme/helpers/integrationPlugin/context.ts (hunk 4 above)
Comment (sethidden) [reply packages/theme/helpers/integrationPlugin/context.ts:1]: Talked on Slack - integrationPlugin is somewhat separate from context (which just shims $sharedRefsMap), so putting it into the context folder wouldn't be right

File: packages/theme/helpers/integrationPlugin/index.ts (hunk 5 above)
Comment (sethidden) [reply packages/theme/helpers/integrationPlugin/index.ts:1]: Talked on Slack - integrationPlugin is somewhat separate from context (which just shims $sharedRefsMap), so putting it into the context folder wouldn't be right

**Review by RetroModernDev — DISMISSED** [review]

**Review by bartoszherba — APPROVED** [review]
